Add introduction to encryption (#3)
icure/medtech-docs: sdks/explanations/encryption/introduction.mdx

---
slug: introduction
---

# Introduction

iCure uses cryptographic algorithms to ensure that only the intended recipients can read sensitive data.

> The iCure SDKs encrypt and decrypt sensitive data on the client-side (end-to-end), this way the sensitive data

Reviewer: ```suggestion
The iCure SDKs encrypt and decrypt sensitive data on the client-side (end-to-end), to insure the sensitive data
```

> travelling over the internet and stored in our servers can't be read by any third party.

Reviewer: ```suggestion
travelling over the internet and stored in our servers cannot be read by any third party.
```

This also includes us: the keys necessary to decrypt the sensitive data are stored only on the client side, therefore we
won't be able to read the sensitive data we receive.

## Symmetric and asymmetric cryptography: basics

The end-to-end encryption scheme in iCure uses standard symmetric (AES-256) and asymmetric (RSA-2048) cryptography
algorithms.

Symmetric cryptography algorithms use a single secret key to encrypt and decrypt the data: if you want to share data

> with other users using symmetric cryptography you will first have to agree on this key.

Reviewer: ```suggestion
with other users using symmetric cryptography, you will first have to agree on this key.
```

> Asymmetric cryptography algorithms instead use pairs of keys, and each pair is composed of a `private key`, which must be

Reviewer: ```suggestion
Asymmetric cryptography algorithms instead use pairs of keys. Each pair is composed of a `private key`, which must be
```
